src/main/java/pl/nogacz/snake/board/Board.java

```diff
@@ -28,22 +29,72 @@
     private int tailLength = 0;
 
     private Coordinates snakeHeadCoordinates = new Coordinates(10, 10);
+    private Coordinates rottenAppleCurrentCoordinates;
+    private Coordinates rottenAppleCurrentCoordinates2;
+    private Coordinates rottenAppleCurrentCoordinates3;
+
+    long lastSpawnTime;
+    long lastDissappearTime;
+
+    long lastSpawnTime2;
+    long lastDissappearTime2;
+
+    long lastSpawnTime3;
+    long lastDissappearTime3;
+
+    int randSpawnRottenAppleTime;
+    int randDissappearRottenAppleTime;
+    
+    int randSpawnRottenAppleTime2;
+    int randDissappearRottenAppleTime2;
+
+    int randSpawnRottenAppleTime3;
+    int randDissappearRottenAppleTime3;
+    
+    
+    boolean disappearRotten = true;
+    boolean disappearRotten2 = true;
+    boolean disappearRotten3 = true;
+
+    
 
     private PawnClass snakeHeadClass = new PawnClass(Pawn.SNAKE_HEAD);
     private PawnClass snakeBodyClass = new PawnClass(Pawn.SNAKE_BODY);
     private PawnClass foodClass = new PawnClass(Pawn.FOOD);
+    
+    private PawnClass rottenAppleClass = new PawnClass(Pawn.ROTTEN_APPLE);
+    private PawnClass rottenAppleClass2 = new PawnClass(Pawn.ROTTEN_APPLE2);
+    private PawnClass rottenAppleClass3 = new PawnClass(Pawn.ROTTEN_APPLE3);
 
     private ArrayList<Coordinates> snakeTail = new ArrayList<>();
+    private ArrayList<Coordinates> rottenApples = new ArrayList<>();
+    private ArrayList<Coordinates> rottenApples2 = new ArrayList<>();
+    private ArrayList<Coordinates> rottenApples3 = new ArrayList<>();
 
     public Board(Design design) {
         this.design = design;
+        
+        randSpawnRottenAppleTime = 5;

```

Comment:
5, 10, 7 ne manaya geliyor?

Bunları birer manalı isimli constanta koymak daha okunaklı olabilir. Ayrıca bunların birimi nedir? Değişken adına birimleri de eklenebilir.

Reply:
blokların ilk spawnlanma zamanlarını belirtiyor. İlk blok 5. saniyede diğeri 7 diğeri 10. saniyede spawnlanıyor.Sonrası tamamıyla random olarak ilerliyor.Sadece initial olarak spawnlanması için böyle. Planladığım algoritmada yalınlık ve kolaylık sağladığı için böyle bir yol tercih ettim.Daha anlamlı bir constant değişkeni ile revize ediyorum hocam.